src/main/scala/viper/gobra/frontend/info/implementation/typing/ExprTyping.scala

```diff
@@ -652,9 +726,17 @@ trait ExprTyping extends BaseTyping { this: TypeInfoImpl =>
       case (Left(_), Some(_: ap.PredExprInstance)) =>
         // a PInvoke on a predicate expression instance must fully apply the predicate arguments
         AssertionT
-      case (Left(callee), Some(_: ap.FunctionCall | _: ap.PredicateCall | _: ap.ClosureCall)) =>
+      case (Left(callee), Some(pattern@(_: ap.FunctionCall | _: ap.PredicateCall | _: ap.ClosureCall))) =>
         exprType(callee) match {
-          case FunctionT(_, res) => res
+          case f: FunctionT => pattern match {
+            case fc: FunctionCall =>
+              // do type inference
+              val typeMap = f.args.filter(isTypeParameter).zip(fc.args.map(exprType)).map {
+                case (TypeParameterT(id, _, _), typ) => (PIdnDef(id.name), typ)
+              }.toMap[PIdnDef, Type]
```

Comment:
I am not sure this is correct. If only the second argument is a type parameter, would you then not zip the second parameter with the first argument?

Reply:
yes you are right, I'll correct this